### PR TITLE
Unite testing of dashboard module

### PR DESCRIPTION
**Pull request by kapilsolaki**

### CODE REVIEW

File: src/app/Modules/auth/pages/login/login.component.ts [hunk 1]
```diff
@@ -43,9 +43,9 @@ export class LoginComponent implements OnInit {
 
   onSubmit(): void {
     // this.submitted = true;
-    if (this.form.valid) {
-      //console.log(JSON.stringify(this.form.value));
-    }
+    // if (this.form.valid) {
```
Comment (teone) [inline src/app/Modules/auth/pages/login/login.component.ts:46]: we should remove the commented out code

**Review by teone — CHANGES_REQUESTED** [review]
I haven't commented on everything but I have a major concern with the implemented unit-tests as they are not checking on the operations they are invoking.

For example:
```
it('should run #ngOnInit()', async () => {
    component.form = component.formBuilder.group({
      email: component.email,
      password: component.password,
    });
    spyOn(component.formBuilder, 'group');
    component.ngOnInit();
    // expect(component.formBuilder.group).toHaveBeenCalled();
  });
```
is setting what `ngOnInit` would set.
A test that validates what `ngOnInit` is expected to do would look like:
```
  it('should run #ngOnInit()', async () => {
    spyOn(component.formBuilder, 'group');
    expect(component.formBuilder.group).toHaveBeenCalled();
    expect(component.form.get("email")).toEqual(component.email)
    expect(component.form.get("password")).toEqual(component.password)
  });
```
_NOTE that we are not calling `component.ngOnInit();` as that is automatically called by Angular itself_

File: src/app/Modules/auth/pages/login/login.component.spec.ts [hunk 2]
```diff
@@ -30,4 +29,35 @@ describe('LoginComponent', () => {
...
+    component.form = component.formBuilder.group({
+      email: component.email,
+      password: component.password,
+    });
+    spyOn(component.formBuilder, 'group');
+    component.ngOnInit();
+    // expect(component.formBuilder.group).toHaveBeenCalled();
+  });
+
+  it('should run #getErrorMessage()', async () => {
```
Comment (teone) [inline src/app/Modules/auth/pages/login/login.component.spec.ts:43]: another thing that I noticed is that all the tests are marked as `async`, while the code looks synchronous, what is the reason for that?

**Comment by SeanCondon** [comment]
retest this please

**Comment by SeanCondon** [comment]
@kapilsolaki - when I get past the License error I'm seeing intermittent Unit test failures with 
```
TypeError: Cannot read properties of undefined (reading '0')
      at SafeSubscriber._next (http://localhost:9876/_karma_webpack_/webpack:/src/app/Modules/dashboard/pages/slices/slices.component.ts:516:3)
```

**Comment by murali-vutti-zensark** [comment]
retest this please

File: src/app/Modules/auth/pages/login/login.component.spec.ts [hunk 3]
```diff
@@ -30,4 +29,31 @@ describe('LoginComponent', () => {
...
+    expect(component.formBuilder.group).toHaveBeenCalled();
+  });
+
+  it('should run #getErrorMessage()', () => {
+    component.email = component.email;
+    spyOn(component.email, 'hasError');
+    component.password = component.password;
+    spyOn(component.password, 'hasError');
+    component.getErrorMessage();
+    expect(component.email.hasError).toHaveBeenCalled();
```
Comment (teone) [inline src/app/Modules/auth/pages/login/login.component.spec.ts:49]: Aside checking that the method is called you might want to check the return value of `getErrorMessage` to make sure it is returning the expected one.

It's not entirely clear to me what `component.email = component.email` is intended to achieve, you're assigning a variable the value it already has.

I believe a test like:
```
    spyOn(component.email, "hasError").and.callThrough();
    spyOn(component.password, "hasError").and.callThrough();
    
    component.email.setValue("foobar");
    component.password.setValue("testpassword");
    
    const res = component.getErrorMessage();
    
    expect(component.email.hasError).toHaveBeenCalledWith('required')
    expect(component.email.hasError).toHaveBeenCalledWith('email')
    expect(component.password.hasError).toHaveBeenCalledWith('required')
    expect(res).toEqual("Not a valid email");
```
will do a better job at validating that the code is working the way it's intended to.

With the current implementation I could the method under test to be:
```
// if (this.email.hasError('required') || this.password.hasError('required')) {
    //   return 'You must enter a value';
    // }
    //
    // return this.email.hasError('email') ? 'Not a valid email' : '';
    this.email.hasError(undefined)
    this.password.hasError(undefined)
    return 'The world is an amazing place'
```
and the test still pass

File: src/app/Modules/dashboard/pages/alerts/alerts.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+  it('should run #setparent()', () => {
+    component.setparent(0);
+  });
+
+  it('should run #setchild()', () => {
+    component.setchild(0);
+  });
+
+  it('should run #selectFilter()', () => {
```
Comment (teone) [inline src/app/Modules/dashboard/pages/alerts/alerts.component.spec.ts:57]: not testing anything

**Review by teone — COMMENTED** [review]
A lot of these tests are still not validating the code. Most don't even have an `expect` clause, the ones that have are only checking that some other method is called, but not that the code is behaving in the expected way.

**Comment by SeanCondon** [comment]
retest this please

**Review by SeanCondon — APPROVED** [review]
Merging this so we can identify any new changes

**Pull request merged**